metricbeat/autodiscover/builder/metrics/metrics_test.go

```diff
@@ -40,48 +41,66 @@ func TestGenerateHints(t *testing.T) {
 			event: bus.Event{
 				"hints": common.MapStr{
 					"metrics": common.MapStr{
-						"module": "prometheus",
+						"module": "mockmodule",
 					},
 				},
 			},
 			len:    0,
 			result: common.MapStr{},
 		},
-		// Only module hint should return empty config
+		// Only module hint should return all metricsets
 		{
 			event: bus.Event{
 				"host": "1.2.3.4",
 				"hints": common.MapStr{
 					"metrics": common.MapStr{
-						"module": "prometheus",
+						"module": "mockmodule",
 					},
 				},
 			},
 			len: 1,
 			result: common.MapStr{
-				"module":     "prometheus",
-				"metricsets": []interface{}{"collector"},
+				"module":     "mockmodule",
+				"metricsets": []interface{}{"one", "two"},
 				"timeout":    "3s",
 				"period":     "1m",
 				"enabled":    true,
 			},
 		},
-		// Only module, namespace hint should return empty config
+		// metricsets hint works
 		{
 			event: bus.Event{
 				"host": "1.2.3.4",
 				"hints": common.MapStr{
 					"metrics": common.MapStr{
-						"module":    "prometheus",
-						"namespace": "test",
+						"module":     "mockmodule",
+						"metricsets": "one",
 					},
 				},
 			},
 			len: 1,
 			result: common.MapStr{
-				"module":     "prometheus",
-				"namespace":  "test",
-				"metricsets": []interface{}{"collector"},
+				"module":     "mockmodule",
+				"metricsets": []interface{}{"one"},
+				"timeout":    "3s",
+				"period":     "1m",
+				"enabled":    true,
+			},
+		},
+		// Only module, it should return defaults
```

Comment:
Instead of a comment here you could add this as `message` or `description` to the test and output it when the test fails. This makes hunting down a failing test much easier. We have that already in quite a few other places.

Reply:
Well, having that this feature is experimental... we could probably live with experimental metricsets being spawned here :innocent: